source/constructs/api/data_source/service.py

```diff
@@ -1383,14 +1384,33 @@ def refresh_aws_data_source(accounts: list[str], type: str):
         elif type == DataSourceType.glue_database.value:
             glue_database_detector.detect(accounts)
 
+        elif type == DataSourceType.jdbc.value:
+            print("jdbc_detector start...")
+            jdbc_detector.detect(Provider.AWS_CLOUD.value, accounts)
         elif type == DataSourceType.all.value:
             s3_detector.detect(accounts)
             rds_detector.detect(accounts)
             glue_database_detector.detect(accounts)
+            jdbc_detector.detect(accounts)
     except Exception as e:
         logger.error(traceback.format_exc())
         raise BizException(MessageEnum.SOURCE_CONNECTION_FAILED.get_code(), str(e))
 
+def refresh_third_data_source(accounts: list[str], type: str):
+    if type is None or len(accounts) == 0:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To check if a container or sequence (string, list, tuple) is empty, use `if not val`. Do not compare its length using `if len(val) == 0`

[Learn more](https://www.python.org/dev/peps/pep-0008/#programming-recommendations#:~:text=if%20not%20seq)